### PR TITLE
[Ubuntu] Add Fastlane

### PR DESCRIPTION
**Pull request by miketimofeev**
# Description
Fastlane is a ruby gem and requires Ruby >= 2.4 thus we don't want to preinstall it everywhere, only on Ubuntu 20 with system Ruby 2.7. However, in this PR, I've added the RubyGems section to each toolset in case we would need to install more gems in the future.
 I also noticed that we don't have tests for system ruby and don't provide RubyGems version in the readme  — added them as well. 
Another point worth to mention — it's a bad idea to update ruby gems from ruby package with `--system` flag and, for example, it's even disabled for Debian so we should either rely on RubyGems from Ruby-full package or install Ruby via other methods.

#### Related issue:
https://github.com/actions/virtual-environments/issues/2676

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

**Review by al-cheb — APPROVED** [review]

File: images/macos/software-report/SoftwareReport.Common.psm1 [hunk 1]
```diff
@@ -181,6 +181,11 @@ function Get-RubyVersion {
     return "Ruby $rubyVersion"
 }
 
+function Get-RubyGemsVersion {
+    $rubyGemsVersion = Run-Command "gem --version"
+    return "RubyGems $rubyGemsVersion"
+}
+
```
Comment (maxim-lobanov) [inline images/macos/software-report/SoftwareReport.Common.psm1:188]: Should this function be in Linux software report? Currently, it is in macOS

Comment (miketimofeev) [reply images/macos/software-report/SoftwareReport.Common.psm1]: Looks like I love macOS more than Linux, fixed

File: images/linux/scripts/installers/ruby.sh [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
-sudo apt-get install ruby-full
-sudo gem update --system
+apt-get install ruby-full
+gem update
+
+# Install ruby gems from toolset
+gemsToInstall=$(get_toolset_value ".rubygems[] .name")
+if [ -n "$gemsToInstall" ]; then
+    for gem in $gemsToInstall; do
+        echo "Installing rubygem $gem"
```
Comment (AlenaSviridenko) [inline images/linux/scripts/installers/ruby.sh:16]: RubyGem is a tool name, the package itself is called gem, so I think the message should be "Installing gem"

Comment (miketimofeev) [reply images/linux/scripts/installers/ruby.sh:16]: makes sense

**Review by AlenaSviridenko — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Pull request merged**